File: 00944. Delete Columns to Make Sorted/00944.c
// https://leetcode.com/problems/delete-columns-to-make-sorted

int minDeletionSize(char ** strs, int size) {
    int i, j, remove = 0;
    int len = strlen(strs[0]);

    for (j = 0; j < len; ++j) {
        for (i = 1; i < size; ++ i) {
            if (strs[i][j] < strs[i-1][j]) {
                ++remove;
                break;
            }
        }
    }

    return remove;
}

File: LeetCode/00944. Delete Columns to Make Sorted/00944.go
// https://leetcode.com/problems/delete-columns-to-make-sorted

func minDeletionSize(strs []string) int {
    var remove, sLen = 0, len(strs[0])

    for idx := 0; idx < sLen; idx++ {
        for s := 1; s < len(strs); s++ {
            if strs[s-1][idx] >  strs[s][idx] {
                remove++
                break
            }
        }
    }

    return remove
}
